### DISCUSSION TITLE
How to render text with better resolution?

### DISCUSSION THREAD

**Discussion by devfservant** [BinaryKits/ZPLUtility#149] [Q&A]
Hello,

We have difficulties in rendering text with a good resolution.
We use `RobotoCondensed-Regular.ttf` font, downloaded from https://fonts.google.com/specimen/Roboto+Condensed.

* Here is what we achieved with BinaryKits.Zpl:
![image](https://user-images.githubusercontent.com/43757707/207824703-fedfffd3-3179-4e44-90be-97f8457bd116.png)
* Here is what we have on Google Fonts website:
![image](https://user-images.githubusercontent.com/43757707/207826745-e7c032ab-4161-40cf-a8b3-9890dd058bdf.png)

When we print the generated document on a zebra printer, it results in a poor readable label.

**How could we improve the quality/resolution of the generated document?**

We tested several things like:

* increasing the `DrawerOptions.RenderQuality` value
* changing the `ZplElementDrawer.draw(elements, labelWidth, labelHeight, printDensityDpmm)` parameters

but it had no effects.

Thank you for your help

**Comment by primo-ppcg** [comment]
The font used for barcode interpretation lines is `A`, be sure this has been set in the FontLoader of the DrawerOptions ([readme](https://github.com/BinaryKits/BinaryKits.Zpl#font-mapping)).

Also, don't send the PNG to the printer, send the ZPL!

EDIT: Could it be that you are drawing the under `^FR` Field Reverse Print? The current implementation will pixelate true type fonts badly, as in the example above.

**Reply by devfservant** [reply]
Thank you for your reply!

Yes, we do use `A` symbols and the `DrawerOptions.FontLoader` option.

One of our requirements is to send PDF content to the printer rather than ZPL text.
Indeed, we build a web application and when you print something through a browser, you have a print preview pop-up
(which cannot be skipped, AFAWK)
Sending ZPL text would generate an "ugly" pop-up for our user, whereas sending a PDF produces a neater pop-up.

We may contribute again to this project to implement a sort of ZplToPdfConverter (as suggested in https://github.com/BinaryKits/BinaryKits.Zpl/issues/115).
Thus, we would send vector data rather than raw images and this would surely fix any resolution issue

**Reply by primo-ppcg** [reply]
Perhaps you could also check if the issue has been resolved with the last commit via the [online viewer](https://binarykits-zpl-viewer.azurewebsites.net/). I noticed that when rendering text in reverse (`^FR`), it became very pixelated.

Before:
![viewer-old](https://user-images.githubusercontent.com/4907320/212277237-f3fefd8f-c651-4b7b-9ee4-6db6bfe526bc.png)

After:
![viewer-new](https://user-images.githubusercontent.com/4907320/212277452-a5d6ad3f-5a98-4a6c-8929-dab69b0b03f3.png)
